packages/backend-core/src/events/events.ts

```diff
@@ -15,6 +45,22 @@ export const publishEvent = async (
   // no backfill - send the event and exit
   if (!backfilling) {
     await processors.processEvent(event, identity, properties, timestamp)
+    if (auditLogsEnabled && isAudited(event)) {
```

Comment:
Best to break this out into an `AuditLogsProcessor` so we don't need to make any changes to the core events publishing - we can just implement a new processor and act there instead. 

We can then move the associated `isAudited`, `configure`, `auditLogsEnabled` and `auditLogQueue` fields there instead so this file is completely agnostic of audit logs.